src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -3025,6 +3025,13 @@ TaskTrackerExecScan(CustomScanState *node)
 	{
 		DistributedPlan *distributedPlan = scanState->distributedPlan;
 		Job *workerJob = distributedPlan->workerJob;
+		Query *jobQuery = workerJob->jobQuery;
+
+		if (ContainsReadIntermediateResultFunction((Node *) jobQuery))
```

Comment:
The only situation in which `ContainsReadIntermediateResultFunction ` returns true here is when you explicitly set `citus.task_executor_type` to `task-tracker` and run a multi-shard query with a CTE that is then replaced by a `read_intermediate_result` call. `read_intermediate_result` errors out when executed by task-tracker, because task-tracker doesn't assign distributed transaction IDs.